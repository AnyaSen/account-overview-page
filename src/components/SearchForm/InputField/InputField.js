import React from "react";
import Styles from "./InputField.module.scss";

export default function InputField({ placeholder, inputValue, onChange }) {
  return (
    <input
      className={Styles.InputField}
      type="text"
      placeholder={placeholder}
      value={inputValue}
      onChange={onChange}
      data-testid="InputField"
    />
  );
}
