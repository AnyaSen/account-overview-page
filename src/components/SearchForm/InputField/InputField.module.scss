@import "../../../assets/scss/variables";
@import "../../../assets/scss/breakpoints";

.InputField {
  padding: $space-1;
  margin: $space-0;
  border: none;
  outline: none;
  background-color: $color-light-grey;
  color: $color-dark-grey;
  font-size: $font-sm;
  border-radius: $base-unit;

  @include lg {
    font-size: $font-sm;
  }

  @include md {
    font-size: $font-sm;
    padding: $space-0 $space-1;
  }
}
