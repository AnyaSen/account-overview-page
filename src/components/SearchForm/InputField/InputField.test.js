import React from "react";
import { render, cleanup } from "@testing-library/react";

import InputField from "./InputField";

describe("<InputField />", () => {
  afterAll(cleanup);

  test("Should have correct placeholder & value.", () => {
    const placeholder = "ID";
    const idValue = "HAS43O";

    const { container, getByPlaceholderText } = render(
      <InputField
        placeholder={placeholder}
        inputValue={idValue}
        onChange={() => {}}
      />
    );

    const input = getByPlaceholderText(placeholder);

    expect(input).toBeTruthy();
    expect(input.value).toBe(idValue);

    expect(container.firstChild).toMatchSnapshot();
  });
});
