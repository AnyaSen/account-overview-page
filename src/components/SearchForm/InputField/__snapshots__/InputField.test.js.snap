// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputField /> Should have correct placeholder & value. 1`] = `
<input
  class="InputField"
  data-testid="InputField"
  placeholder="ID"
  type="text"
  value="HAS43O"
/>
`;
